app/controllers/custom_manifests_controller.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+class CustomManifestsController < SecureController
+  def create
+    manifest = find_manifest
+    if manifest
+      update_manifest(manifest)
+    else
+      create_manifest
+    end
+  end
+
+  def update
+    manifest = find_manifest
+    update_manifest(manifest)
+  end
+
+  def destroy
+    manifest = find_manifest
+    manifest.destroy!
+    head :no_content
+  end
+
+  def show
+    manifest = find_manifest
+    render json: manifest, status: :ok
+  end
+
+  private
+
+  def find_manifest
+    if params[:id]
+      CustomManifest.find(params[:id])
```

Comment:
Should this be `find!`? I don't know the specific use cases here, but I'd think if `params[:id]` was set, but not found, we wouldn't want the controller to create/update and instead raise an error.

Reply:
this should raise, `find` is noisy when a nil or non existing id string : 
(I also always forget which one raises and which one does go silently)
```
2.3.6 :025 > Translation.find(1)
  Translation Load (0.7ms)  SELECT  "translations".* FROM "translations" WHERE "translations"."id" = $1 LIMIT $2  [["id", 1], ["LIMIT", 1]]
 => #<Translation id: 1, is_published: true, version: 1, resource_id: 1, language_id: 1, translated_name: nil, translated_description: nil, manifest_name: nil, translated_tagline: nil> 
2.3.6 :026 > Translation.find(10)
  Translation Load (0.6ms)  SELECT  "translations".* FROM "translations" WHERE "translations"."id" = $1 LIMIT $2  [["id", 10], ["LIMIT", 1]]
ActiveRecord::RecordNotFound: Couldn't find Translation with 'id'=10
	from /opt/local/rvm/gems/ruby-2.3.6@cru/gems/activerecord-5.0.7.2/lib/active_record/core.rb:173:in `find'

```